CMakeLists.txt

```diff
@@ -137,35 +137,30 @@ endif()
 
 # Create a symbolic link from ${base_name} in the binary directory
 # to the corresponding path in the source directory.
+# Note: Copies the file(s) on Windows.
 function(link_to_source base_name)
-    # Get OS dependent path to use in `execute_process`
-    if (CMAKE_HOST_WIN32)
-        #mklink is an internal command of cmd.exe it can only work with \
-        string(REPLACE "/" "\\" link "${CMAKE_CURRENT_BINARY_DIR}/${base_name}")
-        string(REPLACE "/" "\\" target "${CMAKE_CURRENT_SOURCE_DIR}/${base_name}")
-    else()
-        set(link "${CMAKE_CURRENT_BINARY_DIR}/${base_name}")
-        set(target "${CMAKE_CURRENT_SOURCE_DIR}/${base_name}")
-    endif()
+    set(link "${CMAKE_CURRENT_BINARY_DIR}/${base_name}")
+    set(target "${CMAKE_CURRENT_SOURCE_DIR}/${base_name}")
 
     if (NOT EXISTS ${link})
         if (CMAKE_HOST_UNIX)
-            set(command ln -s ${target} ${link})
+            execute_process(COMMAND ln -s ${target} ${link}
+                RESULT_VARIABLE result
+                ERROR_VARIABLE output)
+
+            if (NOT ${result} EQUAL 0)
+                message(FATAL_ERROR "Could not create symbolic link for: ${target} --> ${output}")
+            endif()
         else()
             if (IS_DIRECTORY ${target})
-                set(command cmd.exe /c mklink /j ${link} ${target})
+                file(GLOB_RECURSE files FOLLOW_SYMLINKS LIST_DIRECTORIES false RELATIVE ${target} "${target}/*")
```

Comment:
`file(GLOB_RECURSE` does not have the `LIST_DIRECTORIES` option in cmake 2.28.12, which we must support in 2.28.

Reply:
What can we use instead? I don't know cmake. I can test if I manage to find a cmake 2.28.12 executable for Windows.

Reply:
`LIST_DIRECTORIES false` filters out directories from the list. We can manually check whether each file is a directory or not to predicate whether we link it. I'll put a suggestion shortly.

Reply:
I don't know what's going on, but the code as it stands now is working with `cmake version 2.8.12.2`.

Reply:
If I undesrtand correctly, in CMake 2.8:

* [`file(GLOB_RECURSE …)`](https://cmake.org/cmake/help/v2.8.12/cmake.html#command:file) does not recognize `LIST_DIRECTORIES` so it's treating `LIST_DIRECTORIES` and `false` as extra glob patterns to match, which makes no difference. (And maybe it's also ignoring `RELATIVE ${target}`, but that only mattered when symlinking, not when copying the files.)
* Directories are passed to [`configure_file`](https://cmake.org/cmake/help/v2.8.12/cmake.html#command:configure_file), which causes CMake to know about the directories if they're ever referenced as targets, but they're not.

So it's working by accident, but might cause someone to hate us if they need to do further debugging on it.

Reply:
That's correct I believe. The documentation for `configure_file()` says:
> The <input> must be a file, not a directory.

Which is why the original PR filtered out directories with `LIST_DIRECTORIES`. I'm not sure why this doesn't cause a problem. Perhaps `configure_file()` in 2.8.12 either doesn't care in practice or just ignores directories.

The `RELATIVE` flag also exists in 2.8.12, so that shouldn't be a problem.